libskype.c

```diff
@@ -3491,25 +3492,27 @@ skype_handle_incoming_call(PurpleConnection *gc, char *callnumber_string)
 	{
 		call_media_hash = g_hash_table_new_full(g_str_hash, g_str_equal, g_free, NULL);
 	} else {
-		media = g_hash_table_lookup(call_media_hash, callnumber_string);
+		media = g_hash_table_lookup(call_media_hash, callnumber_str);
 		if (media != NULL)
 		{
 			//we've already dealt with this
 			return;
 		}
 	}
 	
-	temp = skype_send_message("GET CALL %s PARTNER_HANDLE", callnumber_string);
+	temp = skype_send_message("GET CALL %s PARTNER_HANDLE", callnumber_str);
 	if (!temp || !strlen(temp))
 		return;
 	
-	who = g_strdup(&temp[21+strlen(callnumber_string)]);
+	who = g_strdup(&temp[21+strlen(callnumber_str)]);
 	g_free(temp);
 	
 	media = purple_media_manager_create_media(purple_media_manager_get(), purple_connection_get_account(gc), "fsrtpconference", who, FALSE);
 	
 	if (media != NULL)
 	{
+		gchar *callnumber_string = g_strdup(callnumber_str);
```

Comment:
That's the real fix: the `callnumber_str` argument comes from `skype_handle_received_message()`, which will free it when returning, but we need to keep it around (in the hash table, among other things).